fe/fe-core/src/main/java/org/apache/doris/analysis/TimeV2Literal.java

```diff
@@ -0,0 +1,270 @@
+
+public class TimeV2Literal extends LiteralExpr {
+    // part min max store every part of time's min max value
+    public static final TimeV2Literal PART_MIN = new TimeV2Literal(-838, 0, 0, 0, 0);
+    public static final TimeV2Literal PART_MAX = new TimeV2Literal(838, 59, 59, 999999, 6);
+
+    protected long hour;
+    protected long minute;
+    protected long second;
+    protected long microsecond;
+    protected boolean negative;
+
+    /**
+     * C'tor forcing type, e.g., due to implicit cast
+     */
+    // for restore
+    public TimeV2Literal() {
+        this.type = Type.TIMEV2;
+        this.hour = 0;
+        this.minute = 0;
+        this.second = 0;
+        this.microsecond = 0;
+        this.negative = false;
+    }
+
+    public TimeV2Literal(double value) throws AnalysisException {
+        super();
+        if (value > (double) PART_MAX.getValue() || value < -(double) PART_MAX.getValue()) {
+            throw new AnalysisException("The value " + value + " is out of range, expect value range is ["
+                    + (-PART_MAX.getValue()) + ", " + PART_MAX.getValue() + "]");
+        }
+        this.type = ScalarType.createTimeV2Type(6);
+        this.negative = 1.0 / value < 0;
+        long v = (long) Math.abs(value);
+        this.microsecond = (long) (v % 1000000);
+        v /= 1000000;
+        this.second = (long) (v % 60);
+        v /= 60;
+        this.minute = (long) (v % 60);
+        v /= 60;
+        this.hour = (long) v;
+        analysisDone();
+    }
+
+    public TimeV2Literal(long hour, long minute, long second, long microsecond, int scale) throws AnalysisException {
+        super();
+        this.type = ScalarType.createTimeV2Type(scale);
+        this.hour = Math.abs(hour);
+        this.minute = minute;
+        this.second = second;
+        this.microsecond = (long) (microsecond / Math.pow(10, 6 - scale)) * (long) Math.pow(10, 6 - scale);
+        while (microsecond != 0 && this.microsecond < 100000) {
+            this.microsecond *= 10;
+        }
+        this.negative = hour < 0;
+        if (checkRange(this.hour, this.minute, this.second, this.microsecond) || scale > 6 || scale < 0) {
+            throw new AnalysisException("time literal is out of range [-838:59:59.999999, 838:59:59.999999]");
+        }
+        analysisDone();
+    }
+
+    public TimeV2Literal(String s, ScalarType type) {
+        super();
+        this.type = type;
+        init(s);
+        analysisDone();
+    }
+
+    protected TimeV2Literal(TimeV2Literal other) {
+        super(other);
+        this.type = other.type;
+        this.hour = other.getHour();
+        this.minute = other.getMinute();
+        this.second = other.getSecond();
+        this.microsecond = other.getMicroSecond();
+        this.negative = other.isNegative();
+    }
+
+    @Override
+    public Expr clone() {
+        return new TimeV2Literal(this);
+    }
+
+    protected String normalize(String s) {
```

Comment:
remove all analysis in legacy planner's TimeV2Literal. we do not need normalize for legacy planner's literal anymore, because legacy planner has been removed